DeeperFuncTests/Utils.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Utils.swift
+//  DeeperFuncTests
+//
+//  Created by Ilya Puchka on 29/10/2017.
+//  Copyright © 2017 Ilya Puchka. All rights reserved.
+//
+
+import Foundation
+import DeeperFunc
+
+enum Intent: Route, Equatable {
+    
+    case empty
+    case pathAndQueryParams(Int, String, Int, String)
+    case singleParam(Int)
+    case twoParams(Int, String)
+    case anyMiddle
+    case anyEnd
+    case anyStart
+    case anyMiddleParam(Int)
+    case anyMiddleParams(Int, Int)
+    case anyEndParam(Int)
+    case anyStartParam(Int)
+    case orPattern
+    case eitherIntOrInt(Either<Int, Int>)
+    case eitherIntOrString(Either<Int, String>)
+    case eitherIntOrVoid(Either<Int, Void>)
+    case optionalParam(Int?)
+    case optionalSecondParam(Int, String?)
+
+    func deconstruct<A>(_ constructor: ((A) -> Intent)) -> A? {
```

Comment:
This method is used instead of PartialIso that would otherwise be created separately for each case, but will effectively do the same - match specific case with it's constructor and extract associated values.
It's boilerplate, but seems like less code to write comparing with PartialIso.

This code 

`if let values = value.deconstruct(Enum.somecase) {...}` 

is equivalent to this 

`if case let Enum.somecase(values) = value {...}` 

but can be mapped and used without `if`, when standard pattern matching can't be only used in `if case`/`guard case` statements.